x-pack/plugins/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.test.ts

```diff
@@ -462,26 +462,77 @@ describe('The metric threshold alert type', () => {
             },
           ],
           alertOnNoData: true,
+          alertOnGroupDisappear,
         },
         state: state ?? mockOptions.state.wrapped,
       });
     const resultState: any[] = [];
     test('first sends a No Data alert with the * group, but then reports groups when data is available', async () => {
       resultState.push(await execute('test.metric.3'));
       expect(mostRecentAction(instanceID).id).toBe(FIRED_ACTIONS.id);
-      resultState.push(await execute('test.metric.3', resultState.pop()));
+      resultState.push(await execute('test.metric.3', true, resultState.pop()));
       expect(mostRecentAction(instanceID).id).toBe(FIRED_ACTIONS.id);
-      resultState.push(await execute('test.metric.1', resultState.pop()));
+      resultState.push(await execute('test.metric.1', true, resultState.pop()));
       expect(mostRecentAction(instanceID)).toBe(undefined);
       expect(mostRecentAction(instanceIdA).id).toBe(FIRED_ACTIONS.id);
       expect(mostRecentAction(instanceIdB).id).toBe(FIRED_ACTIONS.id);
     });
     test('sends No Data alerts for the previously detected groups when they stop reporting data, but not the * group', async () => {
-      await execute('test.metric.3', resultState.pop());
+      await execute('test.metric.3', true, resultState.pop());
       expect(mostRecentAction(instanceID)).toBe(undefined);
       expect(mostRecentAction(instanceIdA).id).toBe(FIRED_ACTIONS.id);
       expect(mostRecentAction(instanceIdB).id).toBe(FIRED_ACTIONS.id);
     });
+    test('does not send No Data alerts when groups disappear if alertOnGroupDisappear is disabled', async () => {
```

Comment:
Not sure I'm reading this right but I do get a no data alert for * when alertNoData: true and alertOnGroupDisappear is disabled.

Reply:
Yeah it's behaving correctly in this case because *all* groups have disappeared. I can add another data mock where only one of the groups disappears, in order to distinguish these scenarios better.

Reply:
Actually we already had a data mock that produces 3 groups instead of 2. Switched this test case so that it first reports 3 groups, then 2, and ensures that we don't send a No Data alert for the disappeared 3rd group.